llvm/lib/Target/AArch64/AArch64ISelLowering.cpp

```diff
@@ -17840,13 +17887,39 @@ static SDValue performVectorCompareAndMaskUnaryOpCombine(SDNode *N,
   return SDValue();
 }
 
+static SDValue performVectorIntToFpCombine(SDNode *N, SelectionDAG &DAG) {
+  if (N->getOpcode() != ISD::SINT_TO_FP || N->getValueType(0) != MVT::v4f32)
+    return SDValue();
+
+  SDNode *VASHR = N->getOperand(0).getNode();
+  if (VASHR->getOpcode() != AArch64ISD::VASHR ||
```

Comment:
Im not sure if this should be matched from shifts in general. Can we reinstate https://github.com/llvm/llvm-project/pull/91924/files but using fmul and match using that? It would be nice if it could use tablegen like https://reviews.llvm.org/D113199 too, if that is possible.

Reply:
> Im not sure if this should be matched from shifts in general

I think the shift would need to be exact. https://alive2.llvm.org/ce/z/F9usML. From what I can tell we are trying to add this: https://alive2.llvm.org/ce/z/82h6nn, where the shift would be exact, but that wouldn't be true for other cases.

It would probably be good to get the scvtf_fixed working in one patch first with some testing, and have the tbl loop transform in a second.

Reply:
Ack.

Reply:
ISel patterns forked out to https://github.com/llvm/llvm-project/pull/92922 and now require the shift to be exact.